test-data/unit/check-modules.test

```diff
@@ -1640,3 +1640,53 @@ m = n  # E: Cannot assign multiple modules to name 'm' without explicit 'types.M
 [file n.py]
 
 [builtins fixtures/module.pyi]
+
+[case testModuleLevelGetattr]
+import has_getattr
+
+reveal_type(has_getattr.any_attribute)  # E: Revealed type is 'Any'
+
+[file has_getattr.py]
+from typing import Any
+
+def __getattr__(name: str) -> Any: ...
+
+[builtins fixtures/module.pyi]
+
+[case testModuleLevelGetattrReturnType]
+import has_getattr
+
+reveal_type(has_getattr.any_attribute)  # E: Revealed type is 'builtins.str'
+
+[file has_getattr.py]
+def __getattr__(name: str) -> str: ...
+
+[builtins fixtures/module.pyi]
+
+[case testModuleLevelGetattrInvalidSignature]
+import has_getattr
+
+reveal_type(has_getattr.any_attribute)
+
+[file has_getattr.py]
+def __getattr__(x: int, y: str) -> str: ...
+
+[out]
+tmp/has_getattr.py:1: error: Invalid signature "def (builtins.int, builtins.str) -> builtins.str"
+main:3: error: Revealed type is 'builtins.str'
+
+[builtins fixtures/module.pyi]
+
+[case testModuleLevelGetattrNotCallable]
+import has_getattr
+
+reveal_type(has_getattr.any_attribute)  # E: Revealed type is 'Any'
+
+[file has_getattr.py]
+__getattr__ = 3
```

Comment:
I don't think we should allow this to mean `Any`. The text of the PEP specifies a function;
>To make type checkers aware of this, the file can contain the following code:

    def __getattr__(name) -> Any: ...

Furthermore, based on the context of the PEP, I think this should only be allowed in stub files. If people want to `Any` attributes in Python files, it is less intrusive and cleaner imo to have them `# type: ignore` the file, but I think this needs discussion.